src/Project.js

```diff
@@ -82,6 +97,12 @@ export default class Project {
   }
 
   step() {
+    if (this.loudnessHandler.loudness > this._prevLoudness) {
+      this.fireGreatherThanTrigger(Trigger.LOUDNESS_GREATER_THAN);
+    }
+    this._prevLoudness = this.loudnessHandler.loudness;
+    this.fireGreatherThanTrigger(Trigger.TIMER_GREATER_THAN);
```

Comment:
Scratch seems to implement "edge-activated hats" (what we call "greater-than triggers" here) a bit differently--every step, it evaluates the predicate of the hat (e.g. "is loudness greater than N? is timer greater than N?") which returns either a true or false value. It fires those hat blocks if the predicate evaluates to true and it evaluated to false on the previous step (the predicate is on a rising edge).

This seems like it'd require less per-trigger logic, and the predicate logic can be encapsulated more cleanly. Would that be worth doing?